### DISCUSSION TITLE
A/B testing Alerts/Limits

### DISCUSSION THREAD

**Discussion by nairabhishek73** [glific/glific-frontend#2365] [General]
**Is your feature request related to a problem? Please describe.**
When a user employs the split randomly node, they do so for an intended sample size, say 500 people. Currently they have no way of knowing when the required sample has successfully run through the test so that they can stop the test, neither does the node stop the test automatically.

**Describe the solution you'd like**
In an ideal scenario, once the desired sample has run through the A/B testing node, the user gets an alert informing them that the required sample has run through the node and asks them if they would like to redirect users to the default flow. The user should be able to click yes/no and the redirect to the default flow should happen automatically.

**Describe alternatives you've considered**
The alternative would be just sending the alert, informing that sample has been fulfilled and that they should check their analytics for the test and then alter the flow so that the test is no longer active (ie: send users to the default flow)

**Additional context**
You can refer to the KEF support channel for further context: https://discord.com/channels/717975833226248303/971657960726081576/1073090925284302930

**Comment by nairabhishek73** [comment]
@AkhileshNegi @mdshamoon @pankaj-ag

**Comment by mdshamoon** [comment]
Adding the last discussion that we had on this. We will spec out and add support for global variables to potentially solve this.
There are couple of things to be considered while adding global variable support:

- How will it get updated in floweditor
- Where do we define a global variable
- Should it be dynamically or statically typed.

Also we should check this issue in Rapidpro for some insights on this https://github.com/rapidpro/rapidpro/issues/1017

**Comment by pankaj-ag** [comment]
We already have the global variables for flows and some of the NGOs are using it. It's just that those are not exposed in the UI yet.

**Comment by pankaj-ag** [comment]
@glific/glific-frontend-team can you add a new field in the organization setting to update the organization/global fields.
<img width="1113" alt="image" src="https://user-images.githubusercontent.com/10670690/226113227-325a2a1c-0ceb-4310-a08e-5c0242665387.png">

**Comment by mdshamoon** [comment]
We need a writable global field here @pankaj-ag. The usecase is to know how many people passed through a particular node and take action based on that. Right now we have readable global variables which will not help much in this use case.

**Comment by pankaj-ag** [comment]
Okay @mdshamoon  but can't we use collections for this type of use case?  Let's connect sometimes and discuss.

**Comment by mdshamoon** [comment]
Sure let's connect tomorrow after call and discuss this.
